taskcluster/translations_taskgraph/actions/train.py

```diff
@@ -343,6 +330,32 @@ def train_action(parameters, graph_config, input, task_group_id, task_id):
 
     parameters = dict(parameters)
 
+    start_stage = input.pop("start-stage", None)
+    if start_stage:
+        previous_group_ids = input.get("previous_group_ids")
+
+        # First, we create one big graph out of all of the tasks from the specified group IDs.
+        label_to_task_id = {}
+        combined_full_task_graph = {}
+        for graph_id in previous_group_ids:
+            label_to_task_id = get_artifact(graph_id, "public/label-to-taskid.json")
+            full_task_graph = get_artifact(graph_id, "public/full-task-graph.json")
+            combined_full_task_graph.update(full_task_graph)
+        _, combined_full_task_graph = TaskGraph.from_json(combined_full_task_graph)
+
+        # Next, we find the task id(s) corresponding of the tasks that match the stage
+        # we want to start at.
+        start_task_ids = []
+        for label, task in combined_full_task_graph.tasks.items():
+            if task.attributes.get("stage") == start_stage:
+                start_task_ids.append(label_to_task_id[label])
```

Comment:
`label_to_task_id` will be the last thing `get_artifact` returns at the end of `for graph_id in previous_group_ids`. Is that intended? Would it be better to look up `combined_full_task_graph[label]`? We might throw a `KeyError`.

Reply:
Ah, good catch! Yes, this is a subtle bug waiting to happen. I'll fix it.